xmppserver/src/main/webapp/server-session-details.jsp

```diff
@@ -117,7 +122,44 @@
     </tr>
 </tbody>
 </table>
+
 </div>
+<br>
+    <%  // Show Software Version if there is :
+       try {
+        if (!inSessions.get(0).getSoftwareVersion().isEmpty()) {
+    %>
+        <div class="jive-table">
+            <table cellpadding="3" cellspacing="1" border="0" width="100%">
+                <thead>
+                    <tr>
+                        <th colspan="2">
+                            <fmt:message key="session.details.software_version"/>
+                        </th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <% 
+                        Map<String, String> treeMap = new TreeMap<String, String>(inSessions.get(0).getSoftwareVersion());
+                        for (Map.Entry<String, String> entry : treeMap.entrySet()){ %>
+                            <tr>
+                                <td class="c1">
+                                    <%= StringUtils.escapeHTMLTags(entry.getKey().substring(0, 1).toUpperCase()+""+entry.getKey().substring(1)) %>:
+                                </td>
+                                <td>
+                                    <%= StringUtils.escapeHTMLTags(entry.getValue())%>
+                                </td>
+                            </tr>
+                        <% 
+                        }
+                    %>
+                </tbody>
+            </table>
+        </div>
+    <%  } 
+    } catch (Exception e) { 
+       Log.error(e.getMessage(), e);
+    } %>
```

Comment:
It is good to log, but you probably want to show some kind of message to the user looking at the webpage too.